Source/JavaScriptCore/runtime/JSFunction.cpp

```diff
@@ -373,6 +375,12 @@ bool JSFunction::put(JSCell* cell, JSGlobalObject* globalObject, PropertyName pr
 
     JSFunction* thisObject = jsCast<JSFunction*>(cell);
 
+    if (propertyName == vm.propertyNames->displayName) {
```

Comment:
Since `displayName` is a structure property, is there a chance we can implement this change via `ObjectPropertyChangeAdaptiveWatchpoint` is something? I would very much like to avoid adding logic to `JSFunction`'s overrides unless we absolutely have to.